### PR TITLE
feat(Themes): add in G80 theme

### LINKED ISSUE #8471: [element] add gray 80 theme
Add Gray 80 theme for dark inline theme layering modal. See [Notion doc](https://www.notion.so/carbondesignsystem/Final-List-of-new-tokens-fc5985c7331341baa6546064399d36da) for values.

*May have some value tweaks later. I was mostly guessing based on values of the other themes.

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/8471

Adds in the g80 theme for inline theming efforts

#### Changelog

**New**

- `g80` is now available as a theme.

**Changed**

- Added `g80` to the theme switcher in deploy previews

#### Testing / Reviewing

@aagonzales I didn't see a token for `linkPrimaryHover`, I think that's the only one missing

### CODE REVIEW

**Comment by aagonzales** [comment]
@tw15egan Some of the components aren't rendering correctly because its tokens haven't been updated yet. Should we have the v10 tokens values defined for Gray 80? Or will that not matter once we update everything.

File: packages/themes/src/g80.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+// No token listed in notion doc
+export const linkPrimaryHover = blue30;
+
+export const buttonDangerHover = adjustLightness(buttonDangerPrimary, -8);
+export const buttonPrimaryHover = adjustLightness(buttonPrimary, -8);
+export const buttonSecondaryHover = adjustLightness(buttonSecondary, -7);
+export const buttonTertiaryHover = white;
+
+export const backgroundSelected = gray70;
+export const backgroundSelectedHover = adjustLightness(layerSelected, +7);
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: In the preview this is coming up as a much darker gray. 
image

File: packages/themes/src/g80.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+export const focusInverse = blue60;
+export const highlight = blue70;
+
+export const backgroundHover = adjustLightness(background, +6);
+export const layerHover = adjustLightness(layer, +7);
+export const fieldHover = adjustLightness(field, +7);
+export const backgroundInverseHover = adjustLightness(backgroundInverse, -5);
+
+// No token listed in notion doc
+export const linkPrimaryHover = blue30;
```
Comment (aagonzales) [inline packages/themes/src/g80.js:115]: Blue30 is correct

**Comment by tw15egan** [comment]
@aagonzales yeah so I added this one in the opposite way of the other theme updates with new tokens in that I set the new token value to the IDL color and referenced the old tokens to the new ones. 

Will g80 be available as a theme in v10? If not, then we probably don't need to add them if we don't expose the theme on the website or anything

**Review by joshblack — APPROVED** [review]

**Comment by aagonzales** [comment]
No its irrelevant in v10.

**Review by jnm2377 — APPROVED** [review]
🔥

File: packages/themes/src/g80.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+export const icon02 = iconSecondary;
+export const icon03 = iconOnColor;
+
+export const link01 = linkPrimary;
+export const link02 = linkSecondary;
+
+export const inverseLink = linkInverse;
+
+export const field01 = field;
+export const field02 = gray70;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: field02 would actually be Gray60

File: packages/themes/src/g80.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+
+// Interaction states
+export const inverseFocusUi = focusInverse;
+
+export const hoverPrimary = buttonPrimaryHover;
+export const activePrimary = buttonPrimaryActive;
+
+export const hoverPrimaryText = linkPrimaryHover;
+
+export const hoverSecondary = '#606060';
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: hoverSecondary = `buttonSecondaryHover`

File: packages/themes/src/g80.js [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+export const activeSecondary = gray80;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `buttonSecondaryActive`

File: packages/themes/src/g80.js [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+
+export const hoverTertiary = gray10;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `buttonTertiaryHover`

File: packages/themes/src/g80.js [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+export const activeTertiary = gray30;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `buttonTertiaryActive`

File: packages/themes/src/g80.js [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+
+export const hoverUI = backgroundHover;
+export const hoverLightUI = '#656565';
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `#5E5E5E`

File: packages/themes/src/g80.js [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+export const activeUI = backgroundActive;
+export const activeLightUI = gray50;
+export const selectedUI = backgroundSelected;
+export const selectedLightUI = gray60;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `Gray50`

File: packages/themes/src/g80.js [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+export const inverseHoverUI = backgroundInverseHover;
+
+export const hoverSelectedUI = backgroundSelectedHover;
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: Change this to `layerSelectedHover`. In v10 `hoverSelectedUI` works for both background and ui-01, it need ti work for both so use the lighter option.

File: packages/themes/src/g80.js [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+
+export const hoverDanger = buttonDangerHover;
+export const activeDanger = buttonDangerActive;
+
+export const hoverRow = '#4c4c4c';
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `layerHover`

File: packages/themes/src/g80.js (hunk 1 above)
Comment (tw15egan) [reply packages/themes/src/g80.js]: Should it be -7?

**Comment by aagonzales** [comment]
Something still isnt rendering correctly with the selected states. I can't tell if its a value in the theme problem or if its just because the old tokens are being used and it will be fine once the new ones come through. 

image

**Comment by tw15egan** [comment]
`layerSelected` is mapped to `gray60`, so in the new theme, it will be correct. It could be a mismatch between what we had with the old tokens and what we're mapping them to for backwards compatibility for now, I see that`selectedUI` is mapped to `backgroundSelected` (`gray70`). Would mapping `selectedUI` to `layerSelected` possibly fix this?

**Comment by aagonzales** [comment]
Ok lets just leave it alone then. I wont look too hard at the components until they've all been updated with the new tokens.

File: packages/themes/src/g80.js (hunk 1 above)
Comment (aagonzales) [reply packages/themes/src/g80.js]: oh yes that's what it is. should be -7

Comment (tw15egan) [reply packages/themes/src/g80.js]: Fixed!

**Comment by tw15egan** [comment]
@aagonzales I turned on the feature flag so it should all be using the new tokens in all components except the ones in this PR: https://github.com/carbon-design-system/carbon/pull/8427

I'll turn it off before merging but should be the best way to test 👍🏻

File: packages/themes/src/g80.js [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+
+export const linkPrimaryHover = blue30;
+
+export const buttonDangerHover = adjustLightness(buttonDangerPrimary, -8);
+export const buttonPrimaryHover = adjustLightness(buttonPrimary, -8);
+export const buttonSecondaryHover = adjustLightness(buttonSecondary, -7);
+export const buttonTertiaryHover = white;
+
+export const backgroundSelected = gray70;
+export const backgroundSelectedHover = adjustLightness(layerSelected, -7);
```
Comment (aagonzales) [inline packages/themes/src/g80.js]: `backgroundSelected` not `layerSelected`

**Review by aagonzales — COMMENTED** [review]
Thanks TJ! That feature flagged really helped. Just one change I found.

File: packages/themes/src/g80.js (hunk 12 above)
Comment (aagonzales) [reply packages/themes/src/g80.js]: This is my bad, its typed wrong in the notion doc too.

Comment (aagonzales) [reply packages/themes/src/g80.js]: Oops actually it should be `+7` as well. I may have misspoke earlier i think (can't find that previous comment)

**Comment by tw15egan** [comment]
@aagonzales updated 👍🏻

**Review by aagonzales — APPROVED** [review]
🎉  cool! looks good now!

**Pull request merged**